File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/series/model/TvSeriesListResponse.java
package dev.reuss.tmdb.domain.tv.series.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.common.page.PagedResponse;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Paginated TMDB TV series list response.
 *
 * @param page         current page
 * @param results      TV series results
 * @param totalPages   total pages
 * @param totalResults total results
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvSeriesListResponse(
        int page,
        List<TvSeriesListItem> results,

        @JsonProperty("total_pages")
        int totalPages,

        @JsonProperty("total_results")
        int totalResults
) implements PagedResponse<TvSeriesListItem>, TmdbModel {

    public TvSeriesListResponse {
        results = TmdbCollections.list(results);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/series/model/TvSeriesListResponse.kt
package dev.reuss.tmdb.domain.tv.series.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel
import dev.reuss.tmdb.common.page.PagedResponse

/**
 * Paginated TMDB TV series list response.
 *
 * @property page current page
 * @property results TV series results
 * @property totalPages total pages
 * @property totalResults total results
 */
@JvmRecord
data class TvSeriesListResponse(
    @all:JsonProperty("page")
    override val page: Int,

    @all:JsonProperty("results")
    override val results: List<TvSeriesListItem> = emptyList(),

    @all:JsonProperty("total_pages")
    override val totalPages: Int,

    @all:JsonProperty("total_results")
    override val totalResults: Int
) : PagedResponse<TvSeriesListItem>, TmdbModel